### PR TITLE
add test capa presubmit to test boskos integration

### PR DESCRIPTION
**Pull request by Priyankasaggu11929**
The PR adds a test CAPA presubmit job to test the Boskos integration.

Added components:
- Add presubmit on vagator/cluster-api-provider-aws `test-capa-presubmits`
- Update `plugins.yaml` to add `vagator/cluster-api-provider-aws` under the repos.
- Update `config.yaml` to add presets:
    - `preset-kind-volume-mounts: "true"`
    - `preset-service-account: "true"`
- Add test-grid dashboard:`sig-cluster-lifecycle-cluster-api-provider-aws`

### CODE REVIEW

File: config/jobs/cluster-api-provider-aws/presubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    max_concurrency: 1
+    labels:
+      preset-dind-enabled: "true"
+      preset-kind-volume-mounts: "true"
+      preset-service-account: "true"
+      #preset-aws-ssh: "true"
+      preset-aws-credential: "true"
+    spec:
+      containers:
+        - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20220221-c13e827224-master
```
Comment (rajaskakodkar) [inline config/jobs/cluster-api-provider-aws/presubmits.yaml:23]: Lets test this with `public.ecr.aws/t0q8k6g2/vagator/test-image:v20220803-82253a6`

File: config/prow/config.yaml [hunk 2]
```diff
@@ -138,6 +138,44 @@ presets:
...
+        type: Directory
+    - name: cgroup
+      hostPath:
+        path: /sys/fs/cgroup
+        type: Directory
+# credential presets
+- labels:
+    preset-service-account: "true"
+  env:
+  - name: GOOGLE_APPLICATION_CREDENTIALS
```
Comment (rajaskakodkar) [inline config/prow/config.yaml:163]: We won't require this.

File: config/prow/config.yaml [hunk 3]
```diff
@@ -138,6 +138,44 @@ presets:
...
+      hostPath:
+        path: /lib/modules
+        type: Directory
+    - name: cgroup
+      hostPath:
+        path: /sys/fs/cgroup
+        type: Directory
+# credential presets
+- labels:
+    preset-service-account: "true"
```
Comment (rajaskakodkar) [inline config/prow/config.yaml:161]: We won't require this.

Comment (rajaskakodkar) [inline config/prow/config.yaml:161]: We won't require this.

File: config/prow/config.yaml [hunk 4]
```diff
@@ -138,6 +138,44 @@ presets:
     mountPath: /docker-graph
   - name: docker-root
     mountPath: /var/lib/docker
+# volume mounts for kind
+- labels:
+    preset-kind-volume-mounts: "true"
```
Comment (rajaskakodkar) [inline config/prow/config.yaml:143]: Thanks for adding this preset!

Comment (rajaskakodkar) [inline config/prow/config.yaml:143]: Thanks for adding this preset!

File: config/jobs/cluster-api-provider-aws/presubmits.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
+presubmits:
+  vagator/cluster-api-provider-aws:
+  - name: test-capa-presubmits
```
Comment (rajaskakodkar) [inline config/jobs/cluster-api-provider-aws/presubmits.yaml:3]: Let's start integrating Boskos with a TCE AWS job first.

Comment (rajaskakodkar) [inline config/jobs/cluster-api-provider-aws/presubmits.yaml:3]: Let's start integrating Boskos with a TCE AWS job first.

**Review by rajaskakodkar — CHANGES_REQUESTED** [review]
Let's start with a TCE AWS job and then move to capa later

**Comment by Priyankasaggu11929** [comment]
Hey thanks @rajaskakodkar for the review. I got busy in some other stuff & missed redoing it. I'm closing this one, will raise a fresh PR taking in all review comments from above.

**Pull request closed without merging**